apps/signing/views.py

```diff
@@ -81,14 +81,14 @@ def put(self, request, addon, version_string):
                         status=status_code)
 
     @with_addon()
-    def get(self, request, addon, version_string):
+    def get(self, request, addon, version_string, pk=None):
+        file_upload_qs = FileUpload.objects.filter(
+            addon=addon, version=version_string)
         try:
-            file_upload = (FileUpload
-                           .objects
-                           .order_by('-created')
-                           .filter(addon=addon, version=version_string)
-                           [:1]
-                           .get())
+            if pk is None:
```

Comment:
logging for each of these if branches would be nice. Like: `getting latest upload for {addon} {version}: {upload.pk}`